bundles/org.openhab.binding.gridbox/src/main/java/org/openhab/binding/gridbox/internal/GridBoxHandler.java

```diff
@@ -0,0 +1,207 @@
+        updateState(GridBoxBindingConstants.SELF_CONSUMPTION_RATE,
+                new QuantityType<>(newLiveData.getSelfConsumptionRate() * 100, Units.PERCENT));
+        updateState(GridBoxBindingConstants.SELF_SUFFICIENCY_RATE,
+                new QuantityType<>(newLiveData.getSelfSufficiencyRate() * 100, Units.PERCENT));
+        updateState(GridBoxBindingConstants.SELF_SUPPLY, new QuantityType<>(newLiveData.getSelfSupply(), Units.WATT));
+        updateState(GridBoxBindingConstants.TOTAL_CONSUMPTION,
+                new QuantityType<>(newLiveData.getTotalConsumption(), Units.WATT));
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(GridBoxConfiguration.class);
+        if (config.email == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.configuration-error.noemail");
+            return;
+        }
+        if (config.password == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.configuration-error.nopassword");
+            return;
+        }
+
+        updateStatus(ThingStatus.UNKNOWN);
+
+        scheduler.execute(this::initializeApi);
+    }
+
+    private void initializeApi() {
+        updateStatus(ThingStatus.UNKNOWN);
+
+        try {
+            config.idToken = API.getIdToken(config);
+
+            String systemId = config.systemId;
+            if (systemId == null || systemId.isBlank()) {
+                config.systemId = API.getSystemId(config);
+            }
+            updateStatus(ThingStatus.ONLINE);
+
+            updateScheduledFuture = scheduler.scheduleWithFixedDelay(this::update, 0, config.refreshInterval,
+                    TimeUnit.SECONDS);
+        } catch (GridBoxApiAutheticationException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.configuration-error.credentialsinvalid");
+        } catch (IOException | InterruptedException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "@test/offline.communication-error.connectionlost");
+            scheduler.schedule(this::initialize, CONNECTION_RETRY_PERIOD, TimeUnit.SECONDS);
+        } catch (GridBoxApiException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "@test/offline.communication-error.initializeinvalid");
+        }
+    }
+
+    private void update() {
+        try {
+            API.retrieveLiveData(config, this::handleLiveDataResponse);
+            updateStatus(ThingStatus.ONLINE);
+        } catch (GridBoxApiAutheticationException e) {
+            // maybe the authentication is no longer valid, so try to re-authenticate
+            updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.HANDLER_CONFIGURATION_PENDING,
```

Comment:
Unknown should only be set during init, Afterwards it should be offline if an issue occurs. Please change.